Remove ViewEngine from AIO build and CI

alan-agius4 opened the pull request:
This is a pre-update to Angular 12 cleanup.

In version 12 applications cannot be built using View Engine.

alan-agius4 committed:
ci: remove AIO ViewEngine tests

With Angular CLI 12, it is not possible to build ViewEngine applications.

alan-agius4 committed:
build(docs-infra): remove view engine related code

This is a pre-update to Angular 12 cleanup.

This is not needed for version 12, since applications cannot be built using View Engine.

alan-agius4 commented:
@petebacondarwin you also review for `size-tracking`. Thanks

petebacondarwin commented on aio/tools/examples/run-example-e2e.js:
```diff
@@ -180,14 +171,7 @@ function runE2eTestsSystemJS(appDir, outputFile) {
 
-  let run = runProtractorSystemJS(appBuildSpawnInfo.promise, appDir, appRunSpawnInfo, outputFile);
-
-  // Only run AOT tests in ViewEngine mode. The current AOT setup does not work in Ivy.
-  // See https://github.com/angular/angular/issues/35989.
-  if (argv.viewengine && fs.existsSync(appDir + '/aot/index.html')) {
-    run = run.then((ok) => ok && runProtractorAoT(appDir, outputFile));
-  }
```
I think we need to resolve this before removing it. Otherwise there will be ngUpgrade examples that are not being tested.

@gkalpak - do you know if the upgrade guides are in a state to be updated to use AOT?

alan-agius4 replied:
I’ll try to take a look

alan-agius4 replied:
It looks like there is only a single AOT test. 
```
$  ls aio/content/examples/**/aot/index.html 
aio/content/examples/upgrade-phonecat-2-hybrid/aot/index.html
```

petebacondarwin reviewed (approved):
LGTM - but we should get @gkalpak to look at it.

We should also resolve the ngUpgrade testing issue - perhaps before this PR lands?

Reviewed-for: size-tracking

Regarding size-tracking - is there some database of tracking values that will need to be purged after these items are no longer being tracked?

josephperrott reviewed (approved):
LGTM

alan-agius4 committed:
build(docs-infra): update lite-server to fix errors on OSX

```
$ lite-server -c bs-config.aot.json
$ ngc -p tsconfig-aot.json && rollup -c rollup-config.js
/Users/xxxx/git/angular/aio/tools/examples/shared/node_modules/browser-sync/node_modules/chokidar/lib/fsevents-handler.js:26
  return (new fsevents(path)).on('fsevent', callback).start();
          ^
TypeError: fsevents is not a constructor
    at createFSEventsInstance (/Users/xxxx/git/angular/aio/tools/examples/shared/node_modules/browser-sync/node_modules/chokidar/lib/fsevents-handler.js:26:11)
    at setFSEventsListener (/Users/xxxx/git/angular/aio/tools/examples/shared/node_modules/browser-sync/node_modules/chokidar/lib/fsevents-handler.js:80:16)
    at FSWatcher.FsEventsHandler._watchWithFsEvents (/Users/xxxx/git/angular/aio/tools/examples/shared/node_modules/browser-sync/node_modules/chokidar/lib/fsevents-handler.js:244:16)
    at FSWatcher.<anonymous> (/Users/xxxx/git/angular/aio/tools/examples/shared/node_modules/browser-sync/node_modules/chokidar/lib/fsevents-handler.js:378:25)
    at LOOP (fs.js:1752:14)
    at processTicksAndRejections (internal/process/task_queues.js:79:11)
error Command failed with exit code 1.
info Visit https://yarnpkg.com/en/docs/cli/run for documentation about this command.
completed: yarn serve:aot
```

alan-agius4 committed:
test(docs-infra): run all example tests in IVY runtime

We have several tests which use system.js, this means that we cannot use NGCC with `--create-ivy-entry-points`, also we need to process UMD bundles since our system.js configurations use UMDs.

petebacondarwin reviewed (approved):
Nice work Alan - especially solving the ngUpgrade example problem.

alan-agius4 committed:
fixup! test(docs-infra): run all example tests in IVY runtime

AndrewKushnir reviewed (approved):
Reviewed-for: size-tracking

AndrewKushnir commented:
FYI, this changed is merged to master only, since there was a conflict with 12.0.x branch.
